lib/spack/spack/installer.py

```diff
@@ -1165,12 +1172,15 @@ def build_process():
                 spack.compilers.add_compilers_to_config(
                     spack.compilers.find_compilers([pkg.spec.prefix]))
 
-        except StopIteration as e:
+        except spack.build_environment.ChildError as e:
             # A StopIteration exception means that do_install was asked to
-            # stop early from clients.
-            tty.msg('{0} {1}'.format(self.pid, str(e)))
-            tty.msg('Package stage directory : {0}'
-                    .format(pkg.stage.source_path))
+            # stop early from clients, and is not an error at this point
+            if e.name == "StopIteration":
+                tty.debug('{0} {1}'.format(self.pid, str(e)))
+                tty.debug('Package stage directory : {0}'
+                          .format(pkg.stage.source_path))
+            else:
```

Comment:
if you have a new exception type this becomes much cleaner and you can just make it another `except` case.

Reply:
Only tricky bit was making the new type pickleable.